File: Practice/18/Python/task18.py
stop = input("Стоп-слово: ")
words = ["hallo","klempner","das","ist","fantastisch","fluggegecheimen"]
p = {}   # словарь вероятностей букв
a = ord('a')
while (a <= ord('z')):
	p[chr(a)] = 0
	a = a + 1
s = ""
for i in range(len(words)):
	s = s + words[i]
for i in range(len(s)):   # посчитали сколько раз встречается каждая буква
	p[s[i]] += 1
a = ord('a')
while (a <= ord('z')):   # разделили количество каждой буквы на количество всех букв и получили вероятность
	p[chr(a)] /= len(s)
	a = a + 1
prob = 1
for i in range(len(stop)):
	prob *= p[stop[i]]
print ("вероятность равна: " + str(prob))
input("нажмите Enter")

File: Practice/18/C++/task18/task18.cpp
#include <iostream>
#include <map>
#include <string>

using namespace std;

int main() {
	setlocale(LC_ALL, "Russian");
	string words = "hallo" + string("klempner") + "das" + "ist" + "fantastisch" + "fluggegecheimen" ;
	map <char, double> p;
	string stop;
	double prob = 1.0;
	for (char a = 'a'; a <= 'z'; a++) {
		p[a] = 0;
	}
	for (int i = 0; i < size(words); i++) {
		p[words[i]] += 1;
	}
	for (char a = 'a'; a <= 'z'; a++) {
		p[a] = p[a] / size(words);
	}
	cout << "Стоп-слово: ";
	cin >> stop;
	for (int i = 0; i < size(stop); i++) {
		prob = prob * p[stop[i]];
	}
	cout << " Вероятность равна: " << prob << endl;
	return 0;
}